components/PlayerContainer.vue

```diff
@@ -0,0 +1,26 @@
+<template>
+  <PlayerBar :playerSteps="playerSteps" :duration="duration" :extended="isExtendedDuration" @updateModelValue="updateRoutineVariation"/>
+</template>
+<script lang="ts" setup>
+import { onMounted } from "vue";
+import { useRoutineStore } from "@/store/routine";
+const { steps, routineVariation, setRoutineVariation, getRoutineDuration, getPlayerSteps } = useRoutineStore()
```

Comment:
By creating this component, PlayerBar and Player are not using the Pinia Store. They are therefore more isolated and testable, as Vitest break by not being able to use _@pinia-plugin-persistedstate/nuxt_ .